job-server-python/src/python/sparkjobserver/subprocess.py

```diff
@@ -98,6 +100,15 @@ def import_class(cls):
             exit_with_failure(
                     "Expected JavaSparkContext, SQLContext "
                     "or HiveContext but received %s" % repr(context_class), 2)
+
+    egg_path = os.environ.get("EGGPATH", None)
+    if egg_path is not None and sc is not None:
```

Comment:
is empty a valid path?

Also: don't you want to fail if `sc is None`?

Reply:
will change the check for the path.  there are two cases JavaSparkContext and CustomContext that don't appear to set the SC, not sure why for the JavaSparkContext, in those cases the egg won't be added.  depending on the job you have submitted that might be ok.  all of the python tests worked without the egg being added before